Dockerfile

```diff
@@ -3,10 +3,17 @@
 
 FROM ghcr.io/oracle/oraclelinux:9-slim AS jre-build
 
-ENV JAVA_URL="https://download.java.net/java/GA/jdk19.0.2/fdb695a9d9064ad6b064dc6df578380c/7/GPL/openjdk-19.0.2_linux-x64_bin.tar.gz"
+ENV JAVA_URL_X64="https://download.java.net/java/GA/jdk20/bdc68b4b9cbc4ebcb30745c85038d91d/36/GPL/openjdk-20_linux-x64_bin.tar.gz"
+ENV JAVA_URL_AARCH64="https://download.java.net/java/GA/jdk20/bdc68b4b9cbc4ebcb30745c85038d91d/36/GPL/openjdk-20_linux-aarch64_bin.tar.gz"
 
 RUN set -eux; \
     microdnf -y install gzip tar; \
+    MACHINE_TYPE=`uname -m`; \
+    if [ ${MACHINE_TYPE} == 'x86_64' ]; then \
+      JAVA_URL=$JAVA_URL_X64; \
+    else \
```

Comment:
is it safe to assume there is only two types or that the default type is arm?

Reply:
Derek and I tried this on a bunch of machines when we were working on the WKO Dockerfile and this is the same check as over there. We'll certainly need to update this if and when there are more supported machine architectures. Presently, there are only the two choices for Linux binaries: https://jdk.java.net/20/